### ISSUE TITLE
OneZoom site hangs with in a JavaScript infinite loop if tree_startpoints table is empty

### ISSUE THREAD

**Issue by davidebbo** [OneZoom/OZtree#851]
When just going to http://127.0.0.1:8000/OZtree/, the browser hangs with high CPU is the `tree_startpoints` table is empty. This appears to be caused by a bug in `uikit`, but we could potentially work around on our side.

More specifically, in index.html, we have `{{for key in carousel:}}`. If `carousel` is an empty list, we end up with no `<li>` tags, which makes the `uk-slider` blow up.

**Comment by lentinj** [comment]
Wrapping the lot in ``{{if carousel:}}...{{pass}}`` seems like the easy way out. With the same applying for the other carousels in the same template.

**Comment by davidebbo** [comment]
> Wrapping the lot in `{{if carousel:}}...{{pass}}` seems like the easy way out. With the same applying for the other carousels in the same template.

Right, that should do it. I just reported it minimally to uikit as https://github.com/uikit/uikit/issues/5004. Hopefully, if it gets fixed there, we don't need to change our code, other than upgrading uikit.

Getting from the full hang to the minimal repro was ... a fun time!

**Comment by davidebbo** [comment]
The workaround is to simply add a bogus entry to `tree_startpoints`. Values can be mostly random, as long as category is `homepage_main`.

**Comment by davidebbo** [comment]
Good news: the issue was just fixed in uikit. Keeping this open until we update to a uikit with the fix, which will need to be 3.20.9 or later.

**Comment by davidebbo** [comment]
Thanks @lentinj. Agreed, we'll need to do a good round of testing on the UI. Upgrading is not urgent, as I'm unblocked by adding a dummy `tree_startpoints` entry.
